locked-nft/contracts/NFTLocker.cdc

```diff
@@ -147,7 +147,10 @@ pub contract NFTLocker {
                 self.lockedNFTs[nftType] <-! {}
             }
 
-            let oldToken <- self.lockedNFTs.insert(key: nftType, <-{id: <- token})
+            var lockedNftsByType <- self.lockedNFTs.remove(key: nftType)!
```

Comment:
This is going to be an expensive operation if there are many NFTs of the same type locked up

Reply:
lol, just tagged Josh for an alternative

Reply:
i'm aware

Reply:
if you have an alternative also,  please share

Reply:
used a reference as suggested by Josh